src/app/ubs-admin/components/add-violations/add-violations.component.ts

```diff
@@ -0,0 +1,114 @@
+import { Component, OnInit } from '@angular/core';
+import { ImageFile } from '../../models/image-file.model';
+import { FileHandle } from '../../models/file-handle.model';
+import { TranslateService } from '@ngx-translate/core';
+
+@Component({
+  selector: 'app-add-violations',
+  templateUrl: './add-violations.component.html',
+  styleUrls: ['./add-violations.component.scss']
+})
+export class AddViolationsComponent implements OnInit {
+  images: ImageFile[] = [];
+  files: FileHandle[] = [];
+  isImageSizeError = false;
+  isImageTypeError = false;
+  dragAndDropLabel;
+  constructor(private translate: TranslateService) {}
+
+  ngOnInit() {
+    this.translate.get('add-violation-modal.drag-photo').subscribe((value) => {
+      console.log(value);
+      this.dragAndDropLabel = value;
+    });
+    this.initImages();
+  }
+
+  filesDropped(files: FileHandle[]): void {
+    this.isImageSizeError = false;
+    this.isImageTypeError = false;
+    this.checkFileExtension(files);
+  }
+
+  loadFile(event): void {
+    const reader = new FileReader();
+    reader.readAsDataURL(event.target.files[0]);
+
+    reader.onload = () => {
+      this.assignImage(reader.result, event.target.files[0].name);
+    };
+  }
+
+  checkFileExtension(files: FileHandle[]): void {
+    let i = 0;
+    for (const img of this.images) {
+      if (!img.src && files[i]) {
+        if (files[i].file.size <= 10485760) {
+          if (files[i].file.type === 'image/jpeg' || files[i].file.type === 'image/png') {
+            img.src = files[i].url;
+            img.name = files[i].file.name;
+            img.label = null;
+          } else {
+            this.isImageTypeError = true;
+          }
+        } else {
+          this.isImageSizeError = true;
+        }
+        i++;
```

Comment:
this does look good
https://stackoverflow.com/questions/337037/replacing-nested-if-statements
https://medium.com/@jnlewis/how-to-write-clean-deeply-nested-if-else-statements-579d29043f9a

Reply:
done